lint/base_linter/node_linter.py

```diff
@@ -146,21 +147,67 @@ def find_local_executable(self, start_dir, npm_name):
                     self.notify_failure()
                     raise linter.PermanentError()
 
-                # A 'package.json' not yet installed?
-                if not os.path.exists(os.path.join(path, 'node_modules', '.bin')):
-                    is_dep = bool(manifest.get('dependencies', {}).get(npm_name))
-                    is_dev_dep = bool(manifest.get('devDependencies', {}).get(npm_name))
-                    if is_dep or is_dev_dep:
-                        logger.warning(
-                            "Skipping '{}' for now which is listed as a {} "
-                            "in {} but not installed.  Forgot to 'npm install'?"
-                            .format(
-                                npm_name,
-                                'dependency' if is_dep else 'devDependency',
-                                manifest_file
+                is_dep = bool(manifest.get('dependencies', {}).get(npm_name))
+                is_dev_dep = bool(manifest.get('devDependencies', {}).get(npm_name))
+                if is_dep or is_dev_dep:
+                    self.context['project_root'] = path
+
+                    # Perhaps this is a Yarn PnP project?
+                    yarn_path = shutil.which('yarn')
+                    yarn_lock_exists = os.path.exists(os.path.join(path, 'yarn.lock'))
+                    package_lock_exists = os.path.exists(os.path.join(path, 'package-lock.json'))
+                    pnp_js_exists = os.path.exists(os.path.join(path, '.pnp.js'))
+                    pnp_in_manifest = bool(manifest.get('installConfig', {}).get('pnp'))
+
+                    if yarn_path is None:
+                        if yarn_lock_exists and (pnp_in_manifest or pnp_js_exists):
```

Comment:
Why checking for yarn.lock? Isn't a yarn.lock mandatory when you use yarn? Even if if it doesn't exist it would be still a pnp project before the first install?

Reply:
According to **arcanis** [other package managers](https://github.com/SublimeLinter/SublimeLinter/issues/1644#issuecomment-524825828) such as `pnpm` are considering implementing pnp.

Reply:
👍